server/hedley/modules/custom/hedley_admin/scripts/generate-acute-illness-completion-report.php

```diff
@@ -0,0 +1,1052 @@
+      $mode_clause = "AND field_exposure_value IS NOT NULL";
+    }
+    elseif ($mode == "travel_complete") {
+      $mode_clause = "AND field_travel_history_value IS NOT NULL";
+    }
+
+    return db_query("SELECT COUNT (DISTINCT field_acute_illness_encounter_target_id)
+      FROM field_data_field_acute_illness_encounter e
+      LEFT JOIN node ON e.entity_id=node.nid
+      LEFT JOIN field_data_field_individual_participant ip ON e.field_acute_illness_encounter_target_id=ip.entity_id
+      LEFT JOIN field_data_field_person person ON ip.field_individual_participant_target_id=person.entity_id
+      LEFT JOIN field_data_field_health_center hc ON person.field_person_target_id=hc.entity_id
+      LEFT JOIN field_data_field_district district ON person.field_person_target_id=district.entity_id
+      LEFT JOIN field_data_field_ai_encounter_type t ON e.field_acute_illness_encounter_target_id=t.entity_id
+      LEFT JOIN field_data_field_travel_history tra ON node.nid=tra.entity_id
+      LEFT JOIN field_data_field_exposure exp ON node.nid=exp.entity_id
+      WHERE FROM_UNIXTIME(node.created) > :start_date
+        {$name_clause}
+        {$mode_clause}
+        AND FROM_UNIXTIME(node.created) < :end_date",
+      [
+        ':start_date' => $start_date,
+        ':end_date' => $end_date,
+      ])->fetchField();
+  }
+  else {
+    $ids = array_fill_keys(get_id_list($start_date, $end_date, $name_clause), 0);
+
+    $constraints = [
+      "AND field_exposure_value IS NOT NULL",
+      "AND field_travel_history_value IS NOT NULL",
+    ];
+    foreach ($constraints as $constraint_clause) {
+      $results = db_query("SELECT DISTINCT field_acute_illness_encounter_target_id
+      FROM field_data_field_acute_illness_encounter e
+      LEFT JOIN node ON e.entity_id=node.nid
+      LEFT JOIN field_data_field_individual_participant ip ON e.field_acute_illness_encounter_target_id=ip.entity_id
+      LEFT JOIN field_data_field_person person ON ip.field_individual_participant_target_id=person.entity_id
+      LEFT JOIN field_data_field_health_center hc ON person.field_person_target_id=hc.entity_id
+      LEFT JOIN field_data_field_district district ON person.field_person_target_id=district.entity_id
+      LEFT JOIN field_data_field_ai_encounter_type t ON e.field_acute_illness_encounter_target_id=t.entity_id
+      LEFT JOIN field_data_field_travel_history tra ON node.nid=tra.entity_id
+      LEFT JOIN field_data_field_exposure exp ON node.nid=exp.entity_id
+      WHERE FROM_UNIXTIME(node.created) > :start_date
+        {$name_clause}
+        {$constraint_clause}
+        AND FROM_UNIXTIME(node.created) < :end_date",
+      [
+        ':start_date' => $start_date,
+        ':end_date' => $end_date,
+      ])->fetchAll(PDO::FETCH_COLUMN);
+      foreach ($results as $result) {
+        $ids[$result] += 1;
+      }
+    }
+    $count = 0;
+    foreach ($ids as $id) {
+      if ($id == count($constraints)) {
```

Comment:
This is not easily understandable what do we do here. Can we explain it in a comment? Thanks!